cmd/operator/main.go

```diff
@@ -0,0 +1,207 @@
+
+	utilruntime.Must(networkv1alpha1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	var configFile string
+	var timeout string
+	flag.StringVar(&configFile, "config", "",
+		"The controller will load its initial configuration from this file. "+
+			"Omit this flag to use the default configuration values. "+
+			"Command-line flags override configuration from this file.")
+	flag.StringVar(&timeout, "timeout", reconciler.DefaultTimeout,
+		"Timeout for Kubernetes API connections (default: 60s).")
+	opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	options, err := setMangerOptions(configFile)
+	if err != nil {
+		setupLog.Error(err, "error configuring manager options")
+		os.Exit(1)
+	}
+
+	clientConfig := ctrl.GetConfigOrDie()
+	mgr, err := ctrl.NewManager(clientConfig, *options)
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	err = setupReconcilers(mgr, timeout)
+	if err != nil {
+		setupLog.Error(err, "unable to setup reconcilers")
+		os.Exit(1)
+	}
+
+	setupLog.Info("starting manager")
+	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+		setupLog.Error(err, "problem running manager")
+		os.Exit(1)
+	}
+}
+
+func setupReconcilers(mgr manager.Manager, timeout string) error {
+	timoutVal, err := time.ParseDuration(timeout)
+	if err != nil {
+		return fmt.Errorf("error parsing timeout value %s: %w", timeout, err)
+	}
+
+	cr, err := reconciler.NewConfigReconciler(mgr.GetClient(), mgr.GetLogger().WithName("ConfigReconciler"), timoutVal)
+	if err != nil {
+		return fmt.Errorf("unable to create config reconciler reconciler: %w", err)
+	}
+
+	ncr, err := reconciler.NewNodeConfigReconciler(mgr.GetClient(), mgr.GetLogger().WithName("NodeConfigReconciler"), timoutVal)
+	if err != nil {
+		return fmt.Errorf("unable to create node reconciler: %w", err)
+	}
+
+	initialSetup := newOnLeaderElectionEvent(cr)
+	if err := mgr.Add(initialSetup); err != nil {
+		return fmt.Errorf("error adding on leader election event to the manager: %w", err)
+	}
+
+	if err = (&controllers.VRFRouteConfigurationReconciler{
+		Client:     mgr.GetClient(),
+		Scheme:     mgr.GetScheme(),
+		Reconciler: cr,
+	}).SetupWithManager(mgr); err != nil {
+		return fmt.Errorf("unable to create VRFRouteConfiguration controller: %w", err)
+	}
+
+	if err = (&controllers.Layer2NetworkConfigurationReconciler{
+		Client:     mgr.GetClient(),
+		Scheme:     mgr.GetScheme(),
+		Reconciler: cr,
+	}).SetupWithManager(mgr); err != nil {
+		return fmt.Errorf("unable to create Layer2NetworkConfiguration controller: %w", err)
+	}
+
+	if err = (&controllers.RoutingTableReconciler{
+		Client:     mgr.GetClient(),
+		Scheme:     mgr.GetScheme(),
+		Reconciler: cr,
+	}).SetupWithManager(mgr); err != nil {
+		return fmt.Errorf("unable to create RoutingTable controller: %w", err)
+	}
```

Comment:
I think we can just combine all of these into a single controller. That way we also avoid duplicate reconciliations in case multiple kinds of resources are updated at the same time.

Reply:
Done :)